### PR TITLE
add a note about ExtensionSslNativeSupportBuildItem and NativeEnableAllCharsetsBuildItem build items in the extension authors guide

### PR DESCRIPTION
**Pull request by machi1990**
This adds a few sentences about the two build items in the extension author guide.

/cc @gsmet

### CODE REVIEW

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 1]
```diff
@@ -1451,6 +1451,13 @@ A class that will be initialized at runtime rather than build time. This will ca
 `io.quarkus.deployment.builditem.substrate.SubstrateConfigBuildItem`::
 A convenience feature that allows you to control most of the above features from a single build item.
 
+`io.quarkus.deployment.builditem.NativeEnableAllCharsetsBuildItem`::
+Indicates that all charsets should be enabled in native image.
+
+`io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem`::
+A convenience way to tell Quarkus that the extension requires SSL and it should be enable during native image build.
```
Comment (geoand) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: `convinience` -> `convinient`,
`enable` -> `enabled`

File: docs/src/main/asciidoc/extension-authors-guide.adoc [hunk 2]
```diff
@@ -1451,6 +1451,13 @@ A class that will be initialized at runtime rather than build time. This will ca
...
+When using this feature, remember to add your extension in the list of extensions that offers SSL support automatically on the https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/native-and-ssl-guide.adoc[native and ssl guide].
```
Comment (geoand) [inline docs/src/main/asciidoc/extension-authors-guide.adoc]: `in the` -> `to the`
`that offers` -> `that offer`

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 1 above)
Comment (machi1990) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: Good catch. Fixed.

File: docs/src/main/asciidoc/extension-authors-guide.adoc (hunk 2 above)
Comment (machi1990) [reply docs/src/main/asciidoc/extension-authors-guide.adoc]: +1. Thanks for spotting these.

**Review by geoand — APPROVED** [review]
Thanks for the update!

**Pull request merged**